plots/scripts/2A-plot_DEGs.R

```diff
@@ -0,0 +1,203 @@
+#
+# USAGE: Rscript 2A-plot_DEGs.R --cancer_type --subtype_vs_others --subtype_vs_subtype
+
+option_list <- list(
+  optparse::make_option("--cancer_type",
+                        default = NA_character_,
+                        help = "Cancer type"),
+  optparse::make_option("--subtype_vs_others",
+                        help = "Subtype used for comparison against all others."),
+  optparse::make_option("--subtype_vs_subtype",
+                        help = "Subtypes used in head-to-head comparison (comma-separated without space e.g. Type1,Type2)"),
+  optparse::make_option("--proportion_output_directory",
+                        help = "Output directory of DEG proportion plot. Include this option to plot DEG proportion plot."),
+  optparse::make_option("--overlap_output_directory",
+                        help = "Output directory of DEG overlap plot. Include this option to plot silver standard overlap plot."),
+  optparse::make_option("--overlap_measure",
+                        help = "Which overlap measures to include in silver standard overlap plot (comma-separated without space e.g. Jaccard,Rand,Spearman; must be one or more of Jaccard, Rand, Spearman)")
+)
+
+opt <- optparse::parse_args(optparse::OptionParser(option_list=option_list))
+source(here::here("util/option_functions.R"))
+check_options(opt)
+
+# at least one of --subtype_vs_others or --subtype_vs_subtype should be given
+if (any(c("subtype_vs_others", "subtype_vs_subtype") %in% names(opt))) {
+  
+  subtype_vs_others <- NA # first assume option is not provided
+  subtype_vs_subtype <- NA # then update as available below
+  
+  if ("subtype_vs_others" %in% names(opt)) {
+    subtype_vs_others <- opt$subtype_vs_others
+  }
+  
+  if ("subtype_vs_subtype" %in% names(opt)) {
+    subtype_vs_subtype <- opt$subtype_vs_subtype
+  }
+  
+} else {
+  message("  Errors: must include --subtype_vs_others and/or --subtype_vs_subtype in plots/scripts/2A-plot_DEG_proportions.R.\n")
+  stop()
+}
+
+# at least one of --proportion_output_directory or --overlap_output_directory should be given
+if (any(c("proportion_output_directory", "overlap_output_directory") %in% names(opt))) {
+  
+  proportion_output_directory <- NA # first assume option is not provided
+  overlap_output_directory <- NA # then update as available below
+  
+  if ("proportion_output_directory" %in% names(opt)) {
+    proportion_output_directory <- opt$proportion_output_directory
+    plot_proportion <- TRUE
+
+  }
+  
+  if ("overlap_output_directory" %in% names(opt)) {
+    overlap_output_directory <- opt$overlap_output_directory
+    plot_overlap <- TRUE
+    
+    # check that overlap measures requested are the ones present in data
+    if ("overlap_measure" %in% names(opt)) {
+      overlap_measures <- sort(stringr::str_split(opt$overlap_measure,
+                                            pattern = ",", simplify = TRUE))
+      
+      if (!all(overlap_measures %in% c("Jaccard", "Rand", "Spearman"))) {
```

Comment:
Should this be `!any`?

Reply:
Ah I see! I was thinking about this incorrectly / reversing the logic. If you were checking the acceptable measures `%in% overlap_measures`, you could presumably use `any()` and only plot `Jaccard` if `Ran` was supplied. But your way is less permissive and I'd say better.